CMakeLists.txt

```diff
@@ -348,7 +348,7 @@ install(TARGETS st-trace DESTINATION ${CMAKE_INSTALL_BINDIR})
 # Device configuration (Linux only)
 ###
 
-if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND NOT PREVENT_HARDCODED_INSTALL)
```

Comment:
It appears that this does not change anything, nor is it a known, verifiable `cmake` command...

Reply:
It's just an environment variable that, if defined non-null, prevents this install. It works.

In my opinion, it is not a good practice, if a build process of a development tool needs root privileges and forces modifications on the host system. Due to security reason, I never ever run such a build process with root privileges, in particular if you have not reverse-engineered the installation process and know exactly which modifications are done. I'm working on several different projects on the same development host and need to track modification manually. On the other hand a good practice is, if a project can installed relatively for packatizing, archiving, in-docker installation and/or project-local use. The modification on the host system is only done once (fully intentional), but the build process and debugging is done regularly and should never be done with root privileges.
In my environment the stlink stuff are archived in the project together with sources and other resources and the build and installation process of stlink tools are part of the project build. It is installed locally only. 

STLINK_INSTPREFIX       = $(abspath host)
STLINK_CMAKEFLAGS       = -DCMAKE_INSTALL_FULL_DATADIR=$(STLINK_INSTPREFIX) \
                          -DCMAKE_INSTALL_PREFIX=$(STLINK_INSTPREFIX) \
                          -DPREVENT_HARDCODED_INSTALL=1

... make -j1 CMAKEFLAGS="$(STLINK_CMAKEFLAGS)" release install

I agree, that a hobbyist or standard-user may do not need such a feature and for him it would be safer to install everthing in one automated installation process. But there are also experienced users with non-standard use cases.
A compromise and maybe better solution would be, to use the sysconfig_dir feature in you installation script to use it like this:

STLINK_INSTPREFIX       = $(abspath host)
STLINK_CMAKEFLAGS       = -DCMAKE_INSTALL_SYSCONFDIR=$(STLINK_INSTPREFIX)/etc \
                          -DCMAKE_INSTALL_FULL_DATADIR=$(STLINK_INSTPREFIX) \
                          -DCMAKE_INSTALL_PREFIX=$(STLINK_INSTPREFIX)

... make -j1 CMAKEFLAGS="$(STLINK_CMAKEFLAGS)" release install

Reply:
Based on a very similar argumentation, I would not mind either, if a full install would run without root privileges, but in practise we find that this is not possible on linux (and also on windows) for every step of the process due to predefined system default RWX directory permissions.